src/journal/compress.c

```diff
@@ -131,6 +134,29 @@ int compress_blob_lz4(const void *src, uint64_t src_size,
 #endif
 }
 
+int compress_blob_zstd(
+                const void *src, uint64_t src_size,
+                void *dst, size_t dst_alloc_size, size_t *dst_size) {
+#if HAVE_ZSTD
+        size_t k;
+
+        assert(src);
+        assert(src_size > 0);
+        assert(dst);
+        assert(dst_alloc_size > 0);
+        assert(dst_size);
+
+        k = ZSTD_compress(dst, dst_alloc_size, src, src_size, ZSTD_CLEVEL_DEFAULT);
```

Comment:
hmm whhaaa? what a confusing api and documentation about this... I am none the wiser after reading this:

https://facebook.github.io/zstd/zstd_manual.html#Chapter5

So the "DEFAULT" is 3, but 0 is the "default". Got it, make so much sense...

Anyway, will change to 0.

Reply:
With ZSTD_CLEVEL_DEFAULT you bind compression to the definition of that macro while building systemd, Using 0 will leave that to the definition during building libzstd. Theoretically one could build a cut-down version of libzstd which only understands levels 0-2 and this would bug out. Not that I expect this to ever make a difference...